tests/puppeteer/pages/commonPage.js

```diff
@@ -10,4 +10,52 @@ module.exports = class CommonPage {
   async goTo(URL) {
     await this.page.goto(URL);
   }
+
+  /**
+   * Wait for selector and click
+   * @param selector, element to check
+   * @return boolean, true if text exist, false if not
+   */
+  async waitForSelectorAndClick(selector, timeout = 10) {
+    await this.page.waitForSelector(selector, {visible: true, timeout: timeout});
+    await this.page.click(selector)
+  }
+
+  /**
+   * Check text value
+   * @param selector, element to check
+   * @param textToCheckWith, text to check with
+   * @param parameter, parameter to use
+   * @return boolean, true when the result of parameter(textToCheckWith) is true, false if not
+   */
+  async checkTextValue(selector, textToCheckWith, parameter = 'equal') {
+    switch (parameter) {
+      case "equal":
+        await this.page.waitForSelector(selector);
```

Comment:
Maybe doing this before switch and delete it from case 'equal' et 'contain'

Reply:
Yess ! thanks